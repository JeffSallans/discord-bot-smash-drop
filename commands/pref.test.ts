import * as seed from 'seed-random';
seed('123', { global: true });

let importedCommands, importedReadyForCommands, destroy;

describe('preference', () => {
  let mockMessageObject;

  beforeAll(() => {
    const { commands, readyForCommands, destroy } = require('..');
    importedCommands = commands;
    importedReadyForCommands = readyForCommands;
    mockMessageObject = {};
    mockMessageObject.reply = (message) => {
      expect(message).toMatchSnapshot();
    }
  });

  it('setup works correctly', async () => {
    await importedReadyForCommands;
    await importedCommands['setup'](mockMessageObject, ['setup', 'test',
      ':Legendary:', ':Swiggins:', ':Sentry:', ':Yoolip:',
      ':Epic:', ':Coco:', ':Jimmy:', ':Lonestar:', ':MaxFocus:', ':Skolldir:',
      ':Ban:', ':ChuchoKrokk:', ':Derpl:', ':Dizzy:', ':Rocco:', ':Yuri:']);
  });

  it('preference works correctly', async () => {
    await importedReadyForCommands;
    await importedCommands['preference'](mockMessageObject, ['preference', 'test']);
  });

});
