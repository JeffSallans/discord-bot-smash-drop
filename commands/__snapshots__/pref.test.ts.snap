// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preference preference works correctly 1`] = `
"Command: Get Character Preference
test :Legendary: :Swiggins: :Sentry: :Yoolip: :Epic: :Coco: :Jimmy: :Lonestar: :MaxFocus: :Skolldir: :Ban: :ChuchoKrokk: :Derpl: :Dizzy: :Rocco: :Yuri:
"
`;

exports[`preference setup works correctly 1`] = `
"Command: Setup
test preference updated
"
`;
